common/main/java/com/couchbase/lite/internal/fleece/FLEncoder.java

```diff
@@ -228,76 +287,35 @@ public boolean writeValue(@Nullable Object value) {
     }
 
     public boolean write(@Nullable Map<String, Object> map) {
-        if (map == null) { beginDict(0); }
+        boolean ok;
+        if (map == null) { ok = beginDict(0); }
         else {
-            beginDict(map.size());
+            ok = beginDict(map.size());
```

Comment:
If it's not ok, should it return right away?

Reply:
The short-circuit `&&` has the same effect.  I was of the understanding that I had to call `endDict` even if `beginDict` failed, thus the way the call to `endDict` happens before the `&&`

Reply:
Fixed so that endDict will not be called if beginDict does not succeed
Same fix for begin and end Array